rust/noosphere-sphere/src/petname/write.rs

```diff
@@ -137,14 +137,21 @@ where
             return Err(anyhow!("Sphere cannot assign itself to a petname."));
         }
 
+        if let Some(existing_record) = self.get_petname_record(name).await? {
+            if !existing_record.superceded_by(record) {
+                return Err(anyhow!(
```

Comment:
We probably want to make sure that this isn't causing failures where we previously wouldn't have had any during sync: https://github.com/subconsciousnetwork/noosphere/blob/a6f0a74cde2fc001bfff5c1bed0844ac19fc8258/rust/noosphere-sphere/src/sync/strategy.rs#L314

This kind of validation error probably needs to be recovered-from in most cases where it happens. Most likely we have other name records for consideration and one invalid record should not blow up the flow if the others could still be used.

Reply:
Good catch!